store/tikv/mock-tikv/cop_handler_dag.go

```
@@ -0,0 +1,174 @@
+		}
+		chunks = appendRow(chunks, handle, data)
+	}
+	return buildResp(chunks, err)
+}
+
+func (h *rpcHandler) buildExec(ctx *dagContext, curr *tipb.Executor) (executor, error) {
+	var currExec executor
+	switch curr.GetTp() {
+	case tipb.ExecType_TypeTableScan:
+		columns := curr.TblScan.Columns
+		ctx.setColumns(columns)
+		colTps := make(map[int64]*types.FieldType, len(columns))
+		for _, col := range columns {
+			if col.GetPkHandle() {
+				continue
+			}
+			colTps[col.GetColumnId()] = distsql.FieldTypeFromPBColumn(col)
+		}
+		ranges := h.extractKVRanges(ctx.keyRanges, *curr.TblScan.Desc)
+		currExec = &tableScanExec{
+			TableScan: curr.TblScan,
+			kvRanges:  ranges,
+			colTps:    colTps,
+			startTS:   ctx.dagReq.GetStartTs(),
+			mvccStore: h.mvccStore,
+			newRange:  true,
+		}
+	case tipb.ExecType_TypeIndexScan:
+		columns := curr.IdxScan.Columns
+		ctx.setColumns(columns)
+		length := len(columns)
+		// The PKHandle column info has been collected in ctx.
+		if columns[length-1].GetPkHandle() {
+			columns = columns[:length-1]
+		}
+		ids := make([]int64, 0, len(columns))
+		for _, col := range columns {
+			ids = append(ids, col.GetColumnId())
+		}
+		ranges := h.extractKVRanges(ctx.keyRanges, *curr.IdxScan.Desc)
+		currExec = &indexScanExec{
+			IndexScan: curr.IdxScan,
+			kvRanges:  ranges,
+			ids:       ids,
+			startTS:   ctx.dagReq.GetStartTs(),
+			mvccStore: h.mvccStore,
+			newRange:  true,
+		}
+	case tipb.ExecType_TypeSelection:
+		var cond *tipb.Expr
+		// TODO: Now len(Conditions) is 1.
+		if len(curr.Selection.Conditions) > 0 {
+			cond = curr.Selection.Conditions[0]
+		}
+		cols := make(map[int64]*tipb.ColumnInfo)
+		err := extractColumnsInExpr(cond, ctx.columns, cols)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		currExec = &selectionExec{
+			Selection: curr.Selection,
+			eval:      ctx.eval,
+			columns:   cols,
+			sc:        ctx.sc,
+		}
+	default:
+		// TODO: support other types.
```

We should raise an error when meeting the unexpected type.